tests/subsys/net/lib/fota_download/CMakeLists.txt

```diff
@@ -24,16 +24,22 @@ target_include_directories(app
   . # To get 'pm_config.h'
   )
 
+if (NOT CONFIG_TRUSTED_EXECUTION_NONSECURE)
+  # If non secure then fw_info will be pulled in automatically.
+  set(info_magic "-DFIRMWARE_INFO_MAGIC=0xbabababa,0xbabababa,0xbabababa")
+  set(ext_api_magic "-DEXT_API_MAGIC=0xdededede")
+endif()
+
 target_compile_options(app
   PRIVATE
   -DCONFIG_DOWNLOAD_CLIENT_BUF_SIZE=500
   -DCONFIG_DOWNLOAD_CLIENT_STACK_SIZE=500
   -DCONFIG_FW_MAGIC_LEN=32
-  -DFIRMWARE_INFO_MAGIC=0xbabababa,0xbabababa,0xbabababa
   -DABI_INFO_MAGIC=0xdededede
-  -DEXT_API_MAGIC=0xdededede
   -DCONFIG_FW_FIRMWARE_INFO_OFFSET=0x200
   -DCONFIG_FOTA_DOWNLOAD_LOG_LEVEL=2
   -DCONFIG_FOTA_SOCKET_RETRIES=2
   -DCONFIG_FW_INFO_MAGIC_LEN=12
+  ${info_magic}
```

Comment:
Why are this, and all the above config settings handled this way, and not using the `testcase.yml` for handling such settings ?
Like it's done here:
https://github.com/zephyrproject-rtos/zephyr/blob/master/tests/kernel/sched/schedule_api/testcase.yaml#L20-L22
```
    extra_args: CONF_FILE=prj_multiq.conf
    extra_configs:
      - CONFIG_TIMESLICING=n
```
`extra_args` for CMake flags.
`extra_config` for Kconfig symbols.

Reply:
My understanding is that testcase.yml should only be used to separate different test variants (different boards etc) from each other. All of this is common for all test variants.

Reply:
well, all default test case Kconfig settings should normally be in the `prj.conf` file, not in the CMakeLists file to begin with.

This is why you don't see the use of `extra_configs` in testcase.yml when there is only a single variant.

The `extra_args` can be used as soon as you have custom CMake flags, as those should normally not go into the CMakeLists file directly.

So let me rephrase a bit:
Why are this, and all the above config settings handled this way, and not using the prj.conf or testcase.yml for handling such settings ?

Reply:
Because of unfortunate dependencies, kconfig logic and functions being in header files, so that I’m not able to test this feature withouy that level of hard coding. I have tried to do it with «pure» confituration, but the amount of code modifications is not worth it :)

Reply:
Ok, I think we should take a look after 1.5.
There should be some possibilities.

If the problem is with prompt less symbols, have you tried using a dedicate test `Kconfig` defconfig file, like this example:
https://github.com/zephyrproject-rtos/zephyr/blob/master/tests/lib/ringbuffer/Kconfig

Reply:
The biggest problem is the logics of fw_info.h, not kconfig itself. We can take it offline at a later time, I agree :)